Project1/CellEdge.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+class GridCell;
+#include <array>
+#include "MobilityList.h"
+#include <optional>
+/**
+Represents an Edge to a GridCell, basically a struct
+*/
+class CellEdge {
+private:
+	const MobilityList<std::optional<int>> _costs = MobilityList<std::optional<int>>();
+	bool canPass(bool intangible) const;
+public:
+	CellEdge(const CellEdge& cell) = default;
+	CellEdge(CellEdge&& cell) = default;
+
+	GridCell& _cell;
+	CellEdge(GridCell& cell, MobilityList<std::optional<int>> costs);
+	std::optional<int> getCost(MobilityType mobility) const;
+	std::optional<int> getCost(MobilityList<bool> mobility_type) const;
+	std::optional<int> getCost(MobilityList<bool> mobility_type, bool intangible) const;
```

Comment:
Maybe make one that takes a Unit? That way, if there is a Unit in the Target Cell, the Units can talk to see if it can traverse.

Reply:
Makes sense, but wouldn't you need the function to take both units? And the units should have a canPassThrough function in the same place as their movement.

Reply:
The Edge already has the Cell, which has the Tile, which has the Unit* we want to go through. It only needs one Unit.